pkg/serverconf/file.go

```diff
@@ -0,0 +1,100 @@
+package serverconf
+
+import (
+	"errors"
+	"path/filepath"
+	"strconv"
+)
+
+var globalKeys = map[string]bool{
+	"LogPath": true,
+}
+
+type cfg interface {
+	// GlobalMap returns a copy of the top-level (global) configuration map.
+	GlobalMap() map[string]string
+
+	// SubMap returns a copy of the server-specific (if existing) configuration map.
+	SubMap(sub int64) map[string]string
+}
+
+type ConfigFile struct {
+	cfg
+}
+
+func NewConfigFile(path string) (*ConfigFile, error) {
+	var f cfg
+	var err error
+	switch filepath.Ext(path) {
```

Comment:
don't require the config file have a particular extension, just always parse as ini

Reply:
As per #21 I'll still keep the multiple-format boilerplate around (separate `NewConfigFile` from `newinicfg`). Please tell me if you think I should get rid of it as well.